master/internal/api_experiment.go

```diff
@@ -57,16 +60,24 @@ func (a *apiServer) checkExperimentExists(id int) error {
 	}
 }
 
-func (a *apiServer) GetExperiment(
-	_ context.Context, req *apiv1.GetExperimentRequest,
-) (*apiv1.GetExperimentResponse, error) {
+func (a *apiServer) getExperiment(experimentID int) (*experimentv1.Experiment, error) {
 	exp := &experimentv1.Experiment{}
-	switch err := a.m.db.QueryProto("get_experiment", exp, req.ExperimentId); {
+	switch err := a.m.db.QueryProto("get_experiment", exp, experimentID); {
 	case err == db.ErrNotFound:
-		return nil, status.Errorf(codes.NotFound, "experiment not found: %d", req.ExperimentId)
+		return nil, status.Errorf(codes.NotFound, "experiment not found: %d", experimentID)
 	case err != nil:
 		return nil, errors.Wrapf(err,
-			"error fetching experiment from database: %d", req.ExperimentId)
+			"error fetching experiment from database: %d", experimentID)
+	}
+	return exp, nil
```

Comment:
any reason we left the config out of this?

Reply:
I wasn't sure if we want it the user provides the config in the same request. but maybe we add default to it?

Reply:
the old api does provide it

Reply:
I would just add it, personally. The user may be interested with how defaults got filled in, I think that would show?